packages/appframe/src/react/index.tsx

```diff
@@ -202,28 +240,34 @@ function SkipBanner(props) {
   )
 }
 
-SkipBanner.propTypes = {
-  href: PropTypes.string.isRequired
+interface SkipTargetProps extends HTMLAttributes<HTMLAnchorElement> {
+  id: string
 }
-
-const SkipTarget = React.forwardRef((props, ref) => {
-  return <a ref={ref} tabIndex={-1} {...props} />
-})
+const SkipTarget = React.forwardRef<HTMLAnchorElement, SkipTargetProps>(
+  (props, ref) => {
+    return <a ref={ref} tabIndex={-1} {...props} />
+  }
+)
 
 SkipTarget.displayName = 'SkipTarget'
-SkipTarget.propTypes = {
-  id: PropTypes.string.isRequired
-}
 
-function Container(props) {
+const Container: React.FC<HTMLAttributes<HTMLDivElement>> = props => {
   const context = useContext(AppFrameContext)
   return <div {...styles.container(context.sidenavVariant)} {...props} />
 }
 
-function SideNav(props) {
+interface SideNavProps extends HTMLAttributes<HTMLDivElement> {
+  children: AppFrameProps['sidenav']
+}
+interface SideNavStatics {
+  variants: typeof vars.sidenavVariants
+}
+const SideNav: React.FC<SideNavProps> & SideNavStatics = props => {
   const { sidenavVariants: variants } = vars
 
-  const { children, ...rest } = props
+  const { children: _children, ...rest } = props
+  const children = props.children as SideNavProps['children']
+
```

Comment:
@jaketrent so the issue we couldn't fix yesterday is because React.FC appends `{children: ReactNode}` onto the end of whatever props you declare...so i had to get hacky